File: 351004/Lazuta/src/entities/Issue.cpp
#include "Issue.h"

Issue::Issue() : Entity()
{
    m_editorId = 0;
    m_title = "";
    m_content = "";
    m_created = "";
    m_modified = "";
}

Issue::Issue(unsigned long editorId, const std::string& title, 
             const std::string& content, const std::string& created, 
             const std::string& modified)
    : Entity()
{
    m_editorId = editorId;
    m_title = title;
    m_content = content;
    m_created = created;
    m_modified = modified;
}

Issue::~Issue()
{
}

unsigned long Issue::getEditorId() const
{
    return m_editorId;
}

std::string Issue::getTitle() const
{
    return m_title;
}

std::string Issue::getContent() const
{
    return m_content;
}

std::string Issue::getCreated() const
{
    return m_created;
}

std::string Issue::getModified() const
{
    return m_modified;
}

void Issue::setEditorId(unsigned long editorId)
{
    m_editorId = editorId;
}

void Issue::setTitle(const std::string& title)
{
    m_title = title;
}

void Issue::setContent(const std::string& content)
{
    m_content = content;
}

void Issue::setCreated(const std::string& created)
{
    m_created = created;
}

void Issue::setModified(const std::string& modified)
{
    m_modified = modified;
}

File: 351004/Kadushko/Lab1/restApi/src/main/java/com/example/restApi/model/Issue.java
package com.example.restApi.model;

import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

public class Issue extends BaseEntity {
    private Long editorId;
    private String title;
    private String content;
    private LocalDateTime created;
    private LocalDateTime modified;
    private Set<Long> markerIds = new HashSet<>();

    public Long getEditorId() { return editorId; }
    public void setEditorId(Long editorId) { this.editorId = editorId; }

    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }

    public String getContent() { return content; }
    public void setContent(String content) { this.content = content; }

    public LocalDateTime getCreated() { return created; }
    public void setCreated(LocalDateTime created) { this.created = created; }

    public LocalDateTime getModified() { return modified; }
    public void setModified(LocalDateTime modified) { this.modified = modified; }

    public Set<Long> getMarkerIds() { return markerIds; }
    public void setMarkerIds(Set<Long> markerIds) { this.markerIds = markerIds; }
}
